yarn-project/pxe/src/database/kv_pxe_database.ts

```diff
@@ -63,6 +70,28 @@ export class KVPxeDatabase implements PxeDatabase {
     this.#deferredNotesByContract = db.createMultiMap('deferred_notes_by_contract');
   }
 
+  public async addContractArtifact(id: Fr, contract: ContractArtifact): Promise<void> {
+    await this.#contractArtifacts.set(id.toString(), contractArtifactToBuffer(contract));
+  }
+
+  getContractArtifact(id: Fr): Promise<ContractArtifact | undefined> {
+    const contract = this.#contractArtifacts.get(id.toString());
+    // TODO(@spalladino): AztecMap lies and returns Uint8Arrays instead of Buffers, hence the extra Buffer.from.
+    return Promise.resolve(contract && contractArtifactFromBuffer(Buffer.from(contract)));
```

Comment:
It does :smiling_face_with_tear:. A more efficient conversion would be to create a Buffer that refers to the same slice of memory as the UInt8Array
```suggestion
    // TODO(@spalladino): AztecMap lies and returns Uint8Arrays instead of Buffers, hence the extra Buffer.from.
    return Promise.resolve(contract && contractArtifactFromBuffer(Buffer.from(contract.buffer, contract.byteOffset, contract.length / contract.BYTES_PER_ELEMENT)));
    ```
```

As an aside, I remember adding this conversion to the BufferReader (so it'd work with Buffers and UInt8Arrays), but maybe it got removed?

Reply:
To be fair, instead of properly serializing the artifact, we're just json stringifying, so there's a much needed pass to improve performance there!